qemu/tests/chardev_add_port.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Add existed virtio serial port and serial bus
+    1) Start guest with virtio-serial-port and virtio-serial-pci
+    2) execute":"qmp_capabilities"
+    3) Hot-plug existed port1
+    4) Hot plug existed virtio-serial-pci
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    if params['os_type'] == 'windows':
+        session = vm.wait_for_login()
+        check_cmd = params.get("check_vioser_status_cmd",
+                               "verifier /querysettings")
+        output = session.cmd(check_cmd, timeout=360)
+        error_context.context("Make sure vioser.sys verifier enabled in guest.",
+                              logging.info)
+        if "vioser.sys" not in output:
+            verify_cmd = params.get("vioser_verify_cmd",
+                                    "verifier.exe /standard /driver vioser.sys")
+            session.cmd(verify_cmd, timeout=360, ok_status=[0, 2])
+            session = vm.reboot(session=session, timeout=100)
+            output = session.cmd(check_cmd, timeout=360)
+            if "vioser.sys" not in output:
+                test.error("Fail to veirfy vioser.sys driver.")
+
+    assert 'command ignored' in vm.monitor.cmd_obj({"execute": "qmp_capabilities"})['error']['desc']
+    pci_info = vm.monitor.info('pci', debug=False)
+    pci_dev_id = None
+    for dev in pci_info[0]['devices']:
+        print(dev)
+        if dev.get('class_info'):
+            print(dev.get('class_info').get('class'))
+            if dev.get('class_info').get('class') == 1920:
+                pci_dev_id = dev.get('qdev_id')
+                break
+    port = params.objects('serials')[1]
+    virtio_port = vm.devices.get(port)
+    pci_dev = {"driver": "virtio-serial-pci", "max_ports": "31"}
+    pci_dev.update({"id": pci_dev_id})
+    assert 'Duplicate' in vm.monitor.cmd_obj({"execute": "device_add", "arguments": virtio_port.params})['error']['desc']
```

Comment:
`device.hotplug(vm.monitor)`, you can refer `virtio_serial_hotplug_port_pci.py`

Reply:
Hi it is not that useful to use hotplug in this case

Reply:
@michiboo Hi Micky, usually we don't send command directly in case level, but utilize packaged functions instead, for better modularity and scalability.